hx_lib.py

```diff
@@ -331,6 +334,135 @@ def restGetControllerVersion(self):
 
 
 	
+	## IOC Streaming
+	################
+	STREAMING_MODULE_NAME = 'ioc-streaming'
+	STREAMING_MODULE_API_VER = '1'
+
+	# List streaming indicators
+	def restListStreamingIndicators(self, limit=DEFAULT_LIMIT, offset=0, sort_term=None, filter_term={}, query_terms = {}):
+		
+		params = {
+			'limit' : limit,
+			'offset' : offset
+		}
+		if sort_term:
+			params['sort'] = sort_term
+		if filter_term:
+			params['filter'] = filter_term
+		params.update(query_terms)
+		
+		request = self.build_request(self.buildStreamingIndicatorURI(), params = params)
+		(ret, response_code, response_data, _) = self.handle_response(request)
+		
+		return(ret, response_code, response_data)
+
+	# List conditions for a streaming indicator
+	def restListConditionsForStreamingIndicator(self, indicator_id):
+		
+		request = self.build_request(self.buildStreamingIndicatorURI(indicator_id=indicator_id) + '/conditions')
+		(ret, response_code, response_data, _) = self.handle_response(request)
+
+		# filter out deleted conditions
+		not_deleted = list(filter(lambda x: x['deleted'] == False, response_data['data']['entries']))
+		response_data['data']['entries'] = not_deleted
+		return(ret, response_code, response_data)		
+
+	# build the URI for the Indicators endpoint.  Accommodate a specific indicator if provided
+	def buildStreamingIndicatorURI(self, api_endpoint='indicators', indicator_id=None):
```

Comment:
@z3k3z  - just a thought, maybe this would be better as a function decorator for module API calls. As the module name and min API version are going to apply to all modules going forward.

Reply:
I like the idea.